nntrainer/tensor/weight.cpp

```diff
@@ -40,6 +42,29 @@ void Weight::runVariableInitializer() {
   Tensor &var_ref = getVariableRef();
   const TensorDim dim = var_ref.getDim();
 
+  unsigned int fan_in, fan_out;
+
+  /// @fixme: when unit is equal to one, this does not work we need to rely on
+  /// effective dimension then actual numbers here.
+  switch (dim.rank()) {
+  case 1:
+    fan_out = fan_in = dim.width();
+    break;
+  case 2: /// fully connected layers
```

Comment:
if rank == 2, but weight and height are 1 which is true for 1x1 conv filter, IMO this case doesn't handle it correctly.

Reply:
Yes, there is `fixme` right above. Do you have a suggestion instead of relying on effective dimension?

Reply:
aah, I missed that fixme.

Reply:
How about below?
Case 1: B=1, C=1, and rank = 1 - current case 1
Case 2: B=1, C=1, and rank = 2 - current case 2
Default : current case 3

Reply:
Updated :)